pkg/reconciler/ingress/ingress.go

```diff
@@ -209,7 +208,7 @@ func (c *Reconciler) collectLBIngressStatus(ing *v1alpha1.Ingress, gatewayConfig
 					),
 				)
 			}
-			return nil, err
+			return nil, fmt.Errorf("could not find Gateway \"%s/%s\": %w", gwc.Namespace, gwc.Name, err)
```

Comment:
The error here could actually be a network error so I guess we can't prefix "could not find" error message here

Looking at this code some more it seems like after setting `MarkLoadBalancerFailed` we should return _without an error_ because there's no point in retrying to reconcile again.